components/dashboard/src/repositories/detail/workspaces/WorkpaceSizeOptions.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback, useState } from "react";
+import { useWorkspaceClasses } from "../../../data/workspaces/workspace-classes-query";
+import { Label } from "@podkit/forms/Label";
+import { RadioGroup, RadioGroupItem } from "@podkit/forms/RadioListField";
+import { TextMuted } from "@podkit/typography/TextMuted";
+import { Heading2 } from "@podkit/typography/Headings";
+import { ConfigurationSettingsField } from "../ConfigurationSettingsField";
+import { useUpdateProject } from "../../../data/projects/project-queries";
+import { useToast } from "../../../components/toasts/Toasts";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+
+interface Props {
+    configuration: Configuration;
+}
+
+export const ConfigurationWorkspaceSizeOptions = ({ configuration }: Props) => {
+    const [selectedValue, setSelectedValue] = useState(
+        configuration.workspaceSettings?.workspaceClass || "g1-standard",
+    );
+    const classChanged = selectedValue !== configuration.workspaceSettings?.workspaceClass;
+
+    const updateProject = useUpdateProject();
+    const { data: classes, isError, isLoading } = useWorkspaceClasses();
+
+    const { toast } = useToast();
+
+    const setWorkspaceClass = useCallback(
+        async (e: React.FormEvent) => {
+            e.preventDefault();
+
+            updateProject.mutate(
+                {
+                    id: configuration.id,
+                    settings: {
+                        workspaceClasses: {
+                            regular: selectedValue,
+                        },
+                    },
+                },
+                {
+                    onSuccess: () => {
+                        // todo: use optimistic updates when we introduce configuration update hooks
+                        toast({ message: "Workspace size updated" });
+                    },
+                    onError: (e) => {
+                        toast({ message: `Failed updating workspace size: ${e.message}` });
+                    },
+                },
+            );
+        },
+        [configuration.id, selectedValue, toast, updateProject],
+    );
+
+    if (isError) {
+        return <div>Something went wrong</div>;
+    }
+
+    if (isLoading || !classes) {
+        return <div>Loading...</div>;
+    }
+
+    return (
+        <ConfigurationSettingsField className="px-8 py-6">
+            <form onSubmit={setWorkspaceClass}>
+                <div className="mb-4">
+                    <Heading2 asChild>
+                        <h2 className="text-base">Workspace Size Options</h2>
+                    </Heading2>
+                    <TextMuted>Choose the size of your workspace based on the resources you need.</TextMuted>
+                </div>
+                <RadioGroup value={selectedValue} onValueChange={setSelectedValue}>
+                    {classes.map((wsClass) => (
+                        <div className="flex items-start space-x-2 my-2">
```

Comment:
I was thinking it would be nice to use a similar dom hierarchy that I setup for some of our [existing input field components](https://github.com/gitpod-io/gitpod/blob/32aa1bbe79c883ef91ec8f8dd32988495d734a19/components/dashboard/src/components/forms/CheckboxInputField.tsx#L66-L93). In this case, wrapping the input along with the label/description text with the `<label>` has some nice properties (clicking description also toggles input).

I think what you have is fine atm, but I'd like to consider lifting up some more opinionated input field component(s) that give us consistency w/ labels and inputs, as well as field level errors/hints. We can do this in a followup though, but could be helpful to get something in place before we build out a field heavy UI like prebuild settings. That could also be a good time to tackle it as it has a wider variety of input types.

Reply:
Totally agree on the interactive labels. Makes sense to integrate here